ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java

```diff
@@ -75,6 +78,9 @@
   private final KsqlRestClient restClient;
   private final Console terminal;
 
+  private long lastCommandSequenceNumber;
```

Comment:
I would suggest putting both `lastCommandSequenceNumber` and `shouldWaitForPreviousCommand` into a new object, (just an inner class), as both represent state associated with the remote server.  


```java
private static final RemoveServerState{ 
   private long lastCommandSequenceNumber;
   private boolean requestPipelining;

   private RemoveServerState() {
      reset();
   }

   private void reset() {
      lastCommandSequenceNumber = -1L
      requestPipelining = false;
   }
}
```

If/when the user switches to a different KSQL server, (using the `server` command), both of these should be reset.  Having in a separate `ServerState` object, with a reset method that is called when switching servers and in its own the constructor, should ensure a nice pattern for ensuring bad state does not persist when switching servers.